creator-node/src/middlewares/premiumContent/premiumContentMiddleware.test.ts

```diff
@@ -0,0 +1,368 @@
+import { premiumContentMiddleware } from './premiumContentMiddleware'
+import assert from 'assert'
+import { resFactory, loggerFactory } from '../../../test/lib/reqMock'
+import { Request, Response } from 'express'
+import { generateSignature } from '../../apiSigning'
```

Comment:
so that the test can mimic the actual behavior as closely as possible, which in this case means it would generate a signature just like how the DN would

Reply:
kk i don't really see the relevance since they both do the same thing, but doesn't matter

Reply:
I think it's cus in the discovery side, there are other fields passed along with the timestamp. this generate sig fn passes in an extra timestamp in signing data. just used for testing purposes it seems like